ramalama/model.py

```diff
@@ -384,25 +392,52 @@ def check_valid_model_path(self, relative_target_path, model_path):
 
 def get_gpu():
     i = 0
-    gpu_num = 0
-    gpu_bytes = 0
+    amd_gpu_num = -1
+    amd_gpu_bytes = 0
+    nvidia_gpu_num = -1
+    nvidia_gpu_mib = 0
+
+    # Check for AMD GPUs (ROCm/AMD case)
     for fp in sorted(glob.glob('/sys/bus/pci/devices/*/mem_info_vram_total')):
         with open(fp, 'r') as file:
             content = int(file.read())
-            if content > 1073741824 and content > gpu_bytes:
-                gpu_bytes = content
-                gpu_num = i
+            if content > 1073741824 and content > amd_gpu_bytes:
+                amd_gpu_bytes = content
+                amd_gpu_num = i
 
         i += 1
 
-    if gpu_bytes:  # this is the ROCm/AMD case
-        return "HIP_VISIBLE_DEVICES", gpu_num
-
+    # Check if system is running Asahi Linux (Apple Silicon)
     if os.path.exists('/etc/os-release'):
         with open('/etc/os-release', 'r') as file:
             content = file.read()
             if "asahi" in content.lower():
-                return "ASAHI_VISIBLE_DEVICES", 1
+                return "ASAHI_VISIBLE_DEVICES", 1  # For Apple Silicon with Asahi Linux
+
+    # Check for NVIDIA GPUs (CUDA case)
+    try:
+        command = ['nvidia-smi', '--query-gpu=index,memory.total', '--format=csv,noheader,nounits']
```

Comment:
We need to check if nvidia-smi is installed?

Reply:
It is installed on Linux and Windows with the drivers themselves.  I'm not an expert, but I am under the impression it will be present everywhere modern CUDA is usable.

Reply:
Ok well as long as we catch the failure and don't segfault, I will be happy.

Reply:
The final calculation at the end seems overlay complex, if amd_gpu_bytes and nvidia_gpu_mib are set to zero intially. If both are less than 1G VRAM use CPU. Otherwise just use the one with more VRAM.

Reply:
We shouldn't use Nvidia GPUs either with < 1G VRAM. When the VRAM is that small it's not really worth it.

Reply:
Makes sense, I'll rework the function It should be more modular to use other gpu backends (like vulkan, intel arc) down the line

Reply:
I think technically we could only have the kernel module installed outside the container and nvidia-smi installed in the container, the only thing actually required outside of the container is kernelspace stuff, aka kernel modules.

But I think it's ok like this. The detection wont be perfect ever. There will always be corner cases where one must manually specify their container somehow, I was thinking the env vars could be useful for that.

Reply:
Good point! I did add some more modularity for now.

I'm thinking down the road I can query the vulkan SDK for info for all gpu architectures and select from there